### DISCUSSION TITLE
Node input 'X' is not a graph input, initializer, or output of a previous node.

### DISCUSSION THREAD

**Discussion by Onikage3x2** [onnx/tensorflow-onnx#2324] [Q&A]
Hi,

Today, while attempting to use a previously converted Keras model (.h5) to ONNX, specifically a DDRNet 23 Slim model, I encountered this error message. It appears that some inputs were marked as optional, leading to the following issue. Is there a way to set default values for these inputs or designate them as non-optional? I might be on the wrong path with this approach, so any assistance would be greatly appreciated.

Here's the full error message:
`onnxruntime.capi.onnxruntime_pybind11_state.InvalidArgument: [ONNXRuntimeError] : 2 : INVALID_ARGUMENT : Invalid model. Node input 'model_1/add_29/add:0' is not a graph input, initializer, or output of a previous node.`

Additional information:
ONNX version: 1.10
The Keras model was created with TensorFlow 2.5.

Edit:
Here is some more context. 
The error comes after a successfull check: `onnx.checker.check_model(onnx_model)`
In line: `session = onnxruntime.InferenceSession(model_path, providers=['CUDAExecutionProvider', 'CPUExecutionProvider'])`

The input 'model_1/add_29/add:0' not a node, it's a connection.
![Screenshot 2024-04-18 134858](https://github.com/onnx/tensorflow-onnx/assets/20630594/b98b8f7b-7482-4a67-8c2b-fa64439ac8a9)

**Comment by BEEugene** [comment]
Have you solved this issue?
